src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -401,19 +401,21 @@ async function getFlows(
     return validWithinBounds || hasMatchingDeposit;
   });
 
-  const refundRequests: UbaFlow[] = spokePoolClient.getRefundRequests(fromBlock, toBlock).filter((refundRequest) => {
-    const result = refundRequestIsValid(chainId, chainIdIndices, spokePoolClients, hubPoolClient, refundRequest);
-    if (!result.valid && logger !== undefined) {
-      logger.info({
-        at: "UBAClient::getFlows",
-        message: `Excluding RefundRequest on chain ${chainId}`,
-        reason: result.reason,
-        refundRequest,
-      });
-    }
+  const refundRequests: UbaFlow[] = spokePoolClient
```

Comment:
Should we `Promise.all()` these?

Reply:
The `filterAsync` function in utils internally uses this. We will need to rewrite this though b/c this will return a promise.

Reply:
Handled via James' `filterAsync()` suggestion :+1: